L69-cpp-core-event-manager.md

```diff
@@ -0,0 +1,282 @@
+
+The EventManager interface aims at simplifying the usage and maintenance of gRPC event framework. It allows users of gRPC to write their own event management implementation to be used by gRPC for specific needs. The project also paves the way for supporting features such as the C++ callback API in gRPC (See proposal [gRFC L67: C++ callback-based asynchronous API](https://github.com/grpc/proposal/pull/180)).
+
+## Proposal
+The following event manager interfaces will be available in the gRPC core.
+
+```cpp
+// An external handle to an EventManager task, which can be used to cancel the 
+// task. The EventManager task is added via RunFunction.
+struct TaskKey;
+
+// An external handle to an EventManager alarm, which can be used to cancel the
+// alarm. The EventManager alarm is added via AddAlarmIn.
+struct AlarmKey;
+
+// Scheduling priority for closures and alarms added into the EventManager.
+enum Priority {
+  PRIORITY_MIN = 0,
+  PRIORITY_LOW = 0,
+  PRIORITY_MED = 1,
+  PRIORITY_HIGH = 2,
+  PRIORITY_IMMEDIATE = 3,
+  PRIORITY_SYSTEM = 4,
+  PRIORITY_MAX = 4,
+  NUM_PRIORITIES = 5
+};
+
+// A top-level base class of EventManagerInterface.
+class BaseEventManagerInterface {
+ public:
+  virtual ~BaseEventManagerInterface();
+
+  // Adds a new one-shot alarm, which will be scheduled at the deadline to run the 
+  // closure with true. The deadline here is an absolute time. The AlarmKey pointer 
+  // is an output parameter which can be used to cancel the alarm before it fires. 
+  // Note that the closure should not be blocking or long-running, as it would 
+  // block the worker thread in the EventManager from polling the file 
+  // descriptors. In case that the alarm is cancelled successfully in 
+  // ConsistentCancel, the closure will be run with false in the calling thread of 
+  // ConsistentCancel. Callers of ConsistentCancel need to be cautious to avoid 
+  // deadlock.
+  virtual void ScheduleAlarmIn(gpr_timespec deadline, std::function<void(bool)> c, Priority priority, AlarmKey* key = nullptr) = 0;
+
+  // Adds a closure to the EventManager, which will be run at the earliest possible 
+  // time. The TaskKey pointer is an output parameter which can be used to cancel 
+  // the closure at a later time (currently not supported in 
+  // BaseEventManagerInterface). Note that the closure should not be blocking or 
+  // long-running, as it would block the worker thread in the EventManager from 
+  // polling the file descriptors.
```

Comment:
Minor: "file descriptor" in this comment seems out of place. What if the thing being polled isn't a file descriptor?